app/uk/gov/hmrc/gform/formtemplate/DestinationsValidator.scala

```diff
@@ -18,14 +18,17 @@ package uk.gov.hmrc.gform.formtemplate
 
 import cats.data.NonEmptyList
 import uk.gov.hmrc.gform.core.ValidationResult.BooleanToValidationResultSyntax
-import uk.gov.hmrc.gform.core.{ Opt, Valid, ValidationResult }
-import uk.gov.hmrc.gform.sharedmodel.formtemplate.FormTemplate
+import uk.gov.hmrc.gform.core.{ Valid, ValidationResult }
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ FormComponent, FormComponentId, IsGroup }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate.destinations.{ Destination, DestinationId, Destinations }
 
 object DestinationsValidator {
   def someDestinationIdsAreUsedMoreThanOnce(duplicates: Set[DestinationId]) =
     s"Some DestinationIds are defined more than once: ${duplicates.toList.sortBy(_.id).map(_.id)}"
 
+  def groupComponentInDeclaration(groupComponentId: Option[FormComponentId]) =
+    groupComponentId.fold("")(id => s"Group component $id is not allowed in Declaration")
```

Comment:
From reader point of view this looks weird, why is `Option[FormComponentId]` passed here and what empty String as a return value means?

D I bet there is better way how to express this hint. use `fold` instead of `exists` later on?